### PR TITLE
Add docs for outlier detection step.

### PR DESCRIPTION
**Pull request by mairanteodoro**
Resolves [RCAL-503](https://jira.stsci.edu/browse/RCAL-503)

This PR adds the documentation for the outlier detection step.

Successfully ran the regression tests, and the results are available [here](https://plwishmaster.stsci.edu:8081/job/RT/job/Roman-Developers-Pull-Requests/504/).

**Checklist**
- [x] added entry in `CHANGES.rst` under the corresponding subsection
- [ ] updated relevant tests
- [X] updated relevant documentation
- [ ] updated relevant milestone(s)
- [X] added relevant label(s)
- [x] ran regression tests, post a link to the Jenkins job below. [How to run regression tests on a PR](https://github.com/spacetelescope/romancal/wiki/Running-Regression-Tests-Against-PR-Branches)

### CODE REVIEW

**Review by schlafly — COMMENTED** [review]
These look good to me.  There are a few leftover references to JWST that should probably be purged, though it's also probably virtuous to reference that the step's heritage is from Webb.  I forget whether the TSO code remained in the step or not; if not, it probably doesn't belong in the docs.

**Comment by nden** [comment]
@mairanteodoro What is the status of this? Is it ready to be moved out of Draft status?

**Comment by mairanteodoro** [comment]
> @mairanteodoro What is the status of this? Is it ready to be moved out of Draft status?

Yes, it is. Thanks for the reminder, @nden!

File: docs/roman/outlier_detection/main.rst [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+Processing multiple datasets together allows for the identification of bad pixels
+or cosmic-rays that remain in each of the input images, many times at levels which
+were not detectable by the :ref:`jump <jump_step>` step. The ``outlier_detection`` step
+implements the following algorithm to identify and flag any remaining cosmic-rays or
+other artifacts left over from previous calibrations:
+
+  - build a stack of input data
+
+    - all inputs will need to have the same WCS since outlier detection assumes
```
Comment (ddavis-stsci) [inline docs/roman/outlier_detection/main.rst:15]: Can you add links to the step to do this?

File: docs/roman/outlier_detection/main.rst [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      the same flux for each point on the sky, and variations from one image to
+      the next would indicate a problem with the detector during readout of that
+      pixel
```
Comment (ddavis-stsci) [inline docs/roman/outlier_detection/main.rst:18]: I'm not sure that I'd characterize a different distribution of CR as a detector problem.

File: docs/roman/outlier_detection/main.rst [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  - perform a statistical comparison (pixel-by-pixel) between the median blotted
+    data with the original input data to look for pixels with values that are
+    different from the mean value by more than some specified sigma
+    based on the noise model
+
+    - the noise model used relies on the error array computed by previous
+      calibration steps based on the readnoise and calibration errors
+
+  - flag the DQ array for the input data for any pixel (or affected neighboring
+    pixels) identified as a statistical outlier.
```
Comment (ddavis-stsci) [inline docs/roman/outlier_detection/main.rst:38]: I'm not sure what you are saying here. Is there special treatment of neighboring pixels or just that outlying pixels will be flagged even is they are neighbors?

File: docs/roman/outlier_detection/outlier_detection_step.rst [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+OutlierDetectionStep
+--------------------
+
+This module provides the sole interface to all methods of performing outlier detection
+on Roman observations. The outlier detection algorithm used for WFI data is implemented
+in :py:class:`~romancal.outlier_detection.outlier_detection.OutlierDetection`
+and described in :ref:`outlier-detection-imaging`.
+
+.. note::
+    Whether the data are being provided in an ASN file or as a list of ASDF filenames,
```
Comment (ddavis-stsci) [inline docs/roman/outlier_detection/outlier_detection_step.rst:12]: I'd not use the ASN acronym here and would spell it out as association file and link to the asn_from_list docs. 
https://jwst-pipeline.readthedocs.io/en/latest/jwst/associations/asn_from_list.html

File: docs/roman/outlier_detection/outlier_detection.rst [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+
+#. Extract parameter settings from input model and merge them with any user-provided
+   values.
+
+   * See :ref:`outlier detection arguments <outlier_detection_step_args>` for
+     the full list of parameters.
+
+#. By default, resample all input images.
+
+   * The resampling step starts by computing an output WCS that is large enoug
```
Comment (ddavis-stsci) [inline docs/roman/outlier_detection/outlier_detection.rst:20]: enoug -> enough

File: docs/roman/outlier_detection/outlier_detection.rst [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+     to encompass all the input images.
+   * All images from the *same exposure* will get resampled onto this output
+     WCS to create a mosaic of all the chips for that exposure.  This product
```
Comment (ddavis-stsci) [inline docs/roman/outlier_detection/outlier_detection.rst:23]: I think you might mean "All 18 detectors from the same exposure..."?

File: docs/roman/outlier_detection/outlier_detection.rst [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+#. Perform statistical comparison between blotted image and original image to identify outliers.
+
+   * This comparison uses the original input images, the blotted
+     median image, and the derivative of the blotted image to
+     create a cosmic ray mask for each input image.
+   * The derivative of the blotted image gets created using the blotted
+     median image to compute the absolute value of the difference between each pixel and
+     its four surrounding neighbors with the largest value being the recorded derivative.
+   * These derivative images are used to flag cosmic rays
+     and other blemishes, such as satellite trails. Where the difference is larger
```
Comment (ddavis-stsci) [inline docs/roman/outlier_detection/outlier_detection.rst:85]: I'm not sure how many satellite trails we'll be getting at L2, maybe moving object trails?

File: docs/roman/outlier_detection/outlier_examples.rst [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+                }
+
+        .. code-block:: python
+
+                from romancal.outlier_detection import OutlierDetectionStep
+                from romancal.datamodels import ModelContainer
+                # read the file list into a ModelContainer object
+                mc = ModelContainer("asn_sample.json")
+                step = OutlierDetectionStep()
+                step.process(mc)
```
Comment (ddavis-stsci) [inline docs/roman/outlier_detection/outlier_examples.rst:59]: Can you add some command line examples?

**Review by ddavis-stsci — APPROVED** [review]
A few, mostly minor comments, but I think it looks good.

**Review by nden — CHANGES_REQUESTED** [review]
The spec section does not render nicely. The problem was solved in the jwst pipeline, see a PR from Brett

**Comment by mairanteodoro** [comment]
> The spec section does not render nicely. The problem was solved in the jwst pipeline, see a PR from Brett

Done. Applied fix implemented in [PR#8101](https://github.com/spacetelescope/jwst/pull/8101/files).

**Review by nden — APPROVED** [review]

**Pull request merged**